odxtools/compumethods/texttablecompumethod.py

```diff
@@ -28,25 +28,30 @@ def __post_init__(self) -> None:
     def category(self) -> CompuMethodCategory:
         return "TEXTTABLE"
 
-    def _get_scales(self):
+    def _get_scales(self) -> List[CompuScale]:
         scales = list(self.internal_to_phys)
         if self.compu_default_value:
             # Default is last, since it's a fallback
             scales.append(self.compu_default_value)
         return scales
 
-    def convert_physical_to_internal(self, physical_value):
-        scale: CompuScale = next(
-            filter(lambda scale: scale.compu_const == physical_value, self._get_scales()), None)
-        if scale is not None:
-            res = (
-                scale.compu_inverse_value
-                if scale.compu_inverse_value is not None else scale.lower_limit.value)
+    def convert_physical_to_internal(self, physical_value: AtomicOdxType) -> AtomicOdxType:
+        scales = [x for x in self._get_scales() if x.compu_const == physical_value]
+        if scales:
+            inverse_value: Optional[AtomicOdxType] = scales[0].compu_inverse_value
+            if inverse_value is not None:
+                res = inverse_value
+            elif scales[0].lower_limit is not None:
+                res = scales[0].lower_limit.value
+            else:
+                res = None
```

Comment:
Maybe try the other scales? not just scales[0]

Reply:
because of line 39, there should be at most one match (or else the same integer gets mapped to different values). The `scales` variable name is a bit confusing, though...

Reply:
There could be in theory, multiple matching_scales, is there an ODX requirement that says multiple scales can't share the same physical value?

Reply:
if I understood this correctly, that does not make sense for text tables: it would mean that e.g. the value `1` could be mapped to `"ACCELERATE"` as well as to `"BRAKE"`

Reply:
It's the other way around, I know it's kind of an edge case, but if it can happen, it will eventually happen.
```
NO Lower, Upper 1: INVALID
2: ACCELERATE
3: BRAKE
Lower: 4, Upper: 15: INVALID
```

Reply:
okay, I see. fixed: 9ca3ea5acbbcc39c8e8bd76bcb504e04eb0f1c6a

(besides going over the whole list, I also included a check for the upper limit. (i.e., in theory it should now always bail out on the first scale?) )